src/braket/default_simulator/density_matrix_simulation.py

```diff
@@ -149,15 +151,19 @@ def state_with_observables(self) -> np.ndarray:
             raise RuntimeError("No observables applied")
         return self._post_observables
 
+    def expectation(self, with_observables: np.ndarray) -> float:
```

Comment:
Needs comment

Reply:
This overrides `expectation` defined in `Simulation`, and that's where the docs live.